File: backend/src/main/java/com/ieum/admin/festival/domain/model/Festival.java
package com.ieum.admin.festival.domain.model;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 축제 도메인 모델 (순수 자바 객체)
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Festival {

    private Long id;
    private String sourceId;
    
    @Builder.Default
    private FestivalSource source = FestivalSource.MANUAL;
    
    private String title;
    private String description;
    private String overview;
    private String location;
    private String address;
    private LocalDate startDate;
    private LocalDate endDate;
    
    @Builder.Default
    private FestivalStatus status = FestivalStatus.UPCOMING;
    
    private String imageUrl;
    private String thumbnailUrl;
    private String extraImages;
    private String homepage;
    private String tel;
    private Double latitude;
    private Double longitude;
    private String category;
    private String categoryMid;
    private String categorySub;
    private String areaCode;
    private String sigunguCode;
    private String ldongCode;
    private String eventPlace;
    private String playTime;
    private String program;
    private String useFee;
    private String sponsor;
    
    @Builder.Default
    private boolean isCustom = false;
    
    @Builder.Default
    private boolean isVisible = true;
    
    @Builder.Default
    private Double avgRating = 0.0;
    
    @Builder.Default
    private Integer reviewCount = 0;
    
    @Builder.Default
    private Integer favoriteCount = 0;
    
    @Builder.Default
    private Integer viewCount = 0;
    
    private LocalDateTime apiModifiedAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    /**
     * 시작일/종료일 기준으로 축제 상태를 계산하는 도메인 비즈니스 로직
     */
    public static FestivalStatus calculateStatus(java.time.LocalDate startDate, java.time.LocalDate endDate) {
        if (startDate == null || endDate == null) return FestivalStatus.UPCOMING;
        java.time.LocalDate now = java.time.LocalDate.now();
        if (now.isBefore(startDate)) {
            return FestivalStatus.UPCOMING;
        } else if (now.isAfter(endDate)) {
            return FestivalStatus.ENDED;
        } else {
            return FestivalStatus.ONGOING;
        }
    }
}

File: frontend/src/types/festival.ts
export interface Festival {
  id: string;               // 축제 고유 식별자 (PK)
  festivalId?: number;      // 백엔드 API 명세 기준 (숫자 ID)
  title: string;            // 축제 이름
  description?: string;     // 상세 소개 문구
  address: string;          // 개최 현장 상세 주소
  region?: RegionCategory;  // 소속 지역 카테고리
  fee?: number;             // 참가/입장비 (0이면 무료)
  startDate: string;        // 'YYYY-MM-DD'
  endDate: string;          // 'YYYY-MM-DD'
  images?: FestivalImage[]; // 이벤트 썸네일 포스터 이미지 배열
  
  // API 명세 (API_FES_0010) 기준 필드들
  status: 'UPCOMING' | 'ONGOING' | 'ENDED'; // 진행 상황
  imageUrl?: string;
  thumbnailUrl?: string;
  avgStar?: number;
  avgRating?: number;
  reviewCount?: number;
  favoriteCount: number;
  isFavorited: boolean;
  ldongCode?: string;
}

export interface FestivalImage {
  id: string;
  url: string;              // S3 혹은 로컬 CDN URL
  isPrimary: boolean;       // 리스트에 노출될 썸네일 이미지 여부
}

export interface RegionCategory {
  id: number;
  name: string;             // 예: "서울특별시", "강원도"
}
